### PR TITLE
fluent-package 5.0.3

### PR DESCRIPTION
**Pull request by kenhys**
fluent-package 5.0.3

As LTS update, basically updates patch versions.

Exceptions for updated:

* tzinfo-data
  * Timezone database should be updated.

* nokogiri
  * CVE-2024-25062

Exceptions for NOT updated:

* async-http 0.61.0 -> 0.63.0
  * No bug fixes

* oj 3.16.1 -> 3.16.3
  * No need to update because it adds new option and fix for ruby
    3.3.0 or later
* uri 0.12.2 -> 0.13.0
  * ruby 3.2.3 bundles 0.12.2. No CVE furthermore.

Others:

AWS gems are usually updated to the latest every release, but it
is not updated this time to limit the impact.

### CODE REVIEW

**Comment by kenhys** [comment]
nokogiri 1.16 series are not allowed by fluent plugin parser winevt_xml.

**Comment by kenhys** [comment]
fluent-plugin-parser-winevt_xml v0.2.7 has been published, now it can be updated.

**Comment by kenhys** [comment]
still sticked to v1.16.3, need to follow up.

**Comment by kenhys** [comment]
fixed sticked version.

**Comment by kenhys** [comment]
fluent-diagtool  should not abort even though sysctl command is not available. 

https://github.com/fluent/diagtool/pull/21

**Comment by kenhys** [comment]
changed to use $env:TEMP instead of tmp on Windows (c:\fluent\tmp can't be writable)

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,87 @@
...
+## Release v5.0.3 - 2024/03/NN
+
+### News
+
+* Update fluentd to 1.16.4
+* Update bundled gems
+* msi: fixed wrong environment path for Fluent Package Prompt (#606)
+  * It breaks fluent-diagtool behavior to launch fluent-gem correctly.
+* msi: removed unnecessary path delimiter (#607)
+  * It doesn't cause any problem yet, but it should treat `%~dp0~ correctly.
```
Comment (cosmo0920) [inline CHANGELOG.md:14]: Perhaps, did you all forget to add ``"`%~dp0~ with"`` ``"`"`` to emphasize here?

Comment (daipom) [reply CHANGELOG.md:14]: Is this correct?

```suggestion
  * It doesn't cause any problem yet, but it should treat `%~dp0` correctly.
```

Comment (kenhys) [reply CHANGELOG.md]: Thanks! I've fixed it.

**Comment by kenhys** [comment]
need to check  https://github.com/fluent/fluent-package-builder/actions/runs/8324666759/job/22776747302?pr=627 failure.

**Comment by kenhys** [comment]
With fluent-plugin prompt, it works.

```
C:\opt\fluent>fluent-diagtool -t fluentd -o %TEMP%
2024-03-19 10:16:09 +0900: [Diagtool] [INFO] Parsing command options...
2024-03-19 10:16:09 +0900: [Diagtool] [INFO]    Option : Output directory = C:\Users\kenhys\AppData\Local\Temp
2024-03-19 10:16:09 +0900: [Diagtool] [INFO] Initializing parameters...
2024-03-19 10:16:09 +0900: [Diagtool] [INFO] [Collect] Collecting fluent-package gem information...
2024-03-19 10:16:11 +0900: [Diagtool] [INFO] [Collect] fluent-package gem information is stored in C:\Users\kenhys\AppData\Local\Temp/20240319101609/output/tdgem_list.output
2024-03-19 10:16:11 +0900: [Diagtool] [INFO] [Collect] fluent-package gem information (bundled by default) is stored in C:\Users\kenhys\AppData\Local\Temp/20240319101609/output/gem_bundled_list.output
2024-03-19 10:16:11 +0900: [Diagtool] [INFO] [Collect] fluent-package manually installed gem information is stored in C:\Users\kenhys\AppData\Local\Temp/20240319101609/output/gem_local_list.output
2024-03-19 10:16:11 +0900: [Diagtool] [INFO] [Collect] fluent-package manually installed gems:
2024-03-19 10:16:11 +0900: [Diagtool] [INFO] [Collect]   * fluent-plugin-concat
```

**Comment by kenhys** [comment]
Even though invoked from cmd.exe, it works

```
c:\opt\fluent\bin>fluent-diagtool -t fluentd -o %TEMP%
2024-03-19 10:17:55 +0900: [Diagtool] [INFO] Parsing command options...
2024-03-19 10:17:55 +0900: [Diagtool] [INFO]    Option : Output directory = C:\Users\kenhys\AppData\Local\Temp
2024-03-19 10:17:55 +0900: [Diagtool] [INFO] Initializing parameters...
2024-03-19 10:17:55 +0900: [Diagtool] [INFO] [Collect] Collecting fluent-package gem information...
2024-03-19 10:17:56 +0900: [Diagtool] [INFO] [Collect] fluent-package gem information is stored in C:\Users\kenhys\AppData\Local\Temp/20240319101755/output/tdgem_list.output
2024-03-19 10:17:56 +0900: [Diagtool] [INFO] [Collect] fluent-package gem information (bundled by default) is stored in C:\Users\kenhys\AppData\Local\Temp/20240319101755/output/gem_bundled_list.output
2024-03-19 10:17:56 +0900: [Diagtool] [INFO] [Collect] fluent-package manually installed gem information is stored in C:\Users\kenhys\AppData\Local\Temp/20240319101755/output/gem_local_list.output
2024-03-19 10:17:56 +0900: [Diagtool] [INFO] [Collect] fluent-package manually installed gems:
2024-03-19 10:17:56 +0900: [Diagtool] [INFO] [Collect]   * fluent-plugin-concat
```

**Comment by kenhys** [comment]
it was reproduced when set TD_AGENT_TOPDIR= ; set FLUENT_PACKAGE_TOPDIR=.

**Comment by kenhys** [comment]
now CI has passed.

**Comment by kenhys** [comment]
checking on arm.

**Comment by kenhys** [comment]
package can be built on arm:

```
kenhys@M1-Mac-mini f-p-b.work % find fluent-package -name '*.deb'
fluent-package/apt/repositories/debian/pool/bullseye/main/f/fluent-package/fluent-package_5.0.3-1_arm64.deb
fluent-package/apt/repositories/debian/pool/bullseye/main/f/fluent-package/fluent-package-dbgsym_5.0.3-1_arm64.deb
fluent-package/apt/repositories/debian/pool/bullseye/main/f/fluent-package/td-agent_5.0.3-1_all.deb
fluent-package/apt/repositories/debian/pool/bookworm/main/f/fluent-package/fluent-package_5.0.3-1_arm64.deb
fluent-package/apt/repositories/debian/pool/bookworm/main/f/fluent-package/fluent-package-dbgsym_5.0.3-1_arm64.deb
fluent-package/apt/repositories/debian/pool/bookworm/main/f/fluent-package/td-agent_5.0.3-1_all.deb
fluent-package/apt/repositories/ubuntu/pool/focal/universe/f/fluent-package/fluent-package_5.0.3-1_arm64.deb
fluent-package/apt/repositories/ubuntu/pool/focal/universe/f/fluent-package/td-agent_5.0.3-1_all.deb
fluent-package/apt/repositories/ubuntu/pool/jammy/universe/f/fluent-package/fluent-package_5.0.3-1_arm64.deb
fluent-package/apt/repositories/ubuntu/pool/jammy/universe/f/fluent-package/td-agent_5.0.3-1_all.deb
kenhys@M1-Mac-mini f-p-b.work % find fluent-package -name '*.rpm'
fluent-package/yum/repositories/amazon/2023/source/SRPMS/fluent-package-5.0.3-1.amzn2023.src.rpm
fluent-package/yum/repositories/amazon/2023/aarch64/Packages/fluent-package-debugsource-5.0.3-1.amzn2023.aarch64.rpm
fluent-package/yum/repositories/amazon/2023/aarch64/Packages/fluent-package-debuginfo-5.0.3-1.amzn2023.aarch64.rpm
fluent-package/yum/repositories/amazon/2023/aarch64/Packages/fluent-package-5.0.3-1.amzn2023.aarch64.rpm
fluent-package/yum/repositories/amazon/2/source/SRPMS/fluent-package-5.0.3-1.amzn2.src.rpm
fluent-package/yum/repositories/amazon/2/aarch64/Packages/fluent-package-debuginfo-5.0.3-1.amzn2.aarch64.rpm
fluent-package/yum/repositories/amazon/2/aarch64/Packages/fluent-package-5.0.3-1.amzn2.aarch64.rpm
fluent-package/yum/repositories/rocky/8/source/SRPMS/fluent-package-5.0.3-1.el8.src.rpm
fluent-package/yum/repositories/rocky/8/aarch64/Packages/fluent-package-debuginfo-5.0.3-1.el8.aarch64.rpm
fluent-package/yum/repositories/rocky/8/aarch64/Packages/fluent-package-debugsource-5.0.3-1.el8.aarch64.rpm
fluent-package/yum/repositories/rocky/8/aarch64/Packages/fluent-package-5.0.3-1.el8.aarch64.rpm
fluent-package/yum/repositories/centos/7/source/SRPMS/fluent-package-5.0.0-1.el7.src.rpm
fluent-package/yum/repositories/centos/7/source/SRPMS/fluent-package-5.0.3-1.el7.src.rpm
fluent-package/yum/repositories/centos/7/aarch64/Packages/fluent-package-debuginfo-5.0.3-1.el7.aarch64.rpm
fluent-package/yum/repositories/centos/7/aarch64/Packages/fluent-package-debuginfo-5.0.0-1.el7.aarch64.rpm
fluent-package/yum/repositories/centos/7/aarch64/Packages/fluent-package-5.0.0-1.el7.aarch64.rpm
fluent-package/yum/repositories/centos/7/aarch64/Packages/fluent-package-5.0.3-1.el7.aarch64.rpm
fluent-package/yum/repositories/almalinux/9/source/SRPMS/fluent-package-5.0.3-1.el9.src.rpm
fluent-package/yum/repositories/almalinux/9/aarch64/Packages/fluent-package-5.0.3-1.el9.aarch64.rpm
fluent-package/yum/repositories/almalinux/9/aarch64/Packages/fluent-package-debugsource-5.0.3-1.el9.aarch64.rpm
fluent-package/yum/repositories/almalinux/9/aarch64/Packages/fluent-package-debuginfo-5.0.3-1.el9.aarch64.rpm
```

**Comment by kenhys** [comment]
OT: downloading artifacts via script may stall. checking why...

**Comment by kenhys** [comment]
@daipom could you merge this PR?

**Review by daipom — APPROVED** [review]
LGTM! Thanks!

**Comment by kenhys** [comment]
Uploaded to test/experimental:

OK: debian:bullseye exp/5
OK: debian:bullseye exp/lts/5
OK: debian:bookworm exp/5
OK: debian:bookworm exp/lts/5
OK: ubuntu:focal exp/5
OK: ubuntu:focal exp/lts/5
OK: ubuntu:jammy exp/5
OK: ubuntu:jammy exp/lts/5
OK: centos:7 exp/5
OK: centos:7 exp/lts/5
OK: almalinux:8 exp/5
OK: almalinux:8 exp/lts/5
OK: rockylinux:9 exp/5
OK: rockylinux:9 exp/lts/5
OK: amazonlinux:2 exp/5
OK: amazonlinux:2 exp/lts/5
OK: amazonlinux:2023 exp/5
OK: amazonlinux:2023 exp/lts/5

**Pull request merged**